target/linux/mediatek/mt7622/base-files/lib/upgrade/platform.sh

```diff
@@ -75,6 +75,9 @@ platform_check_image() {
 	xiaomi,redmi-router-ax6s)
 		nand_do_platform_check "$board" "$1"
 		;;
+	netgear,wax206)
+		nand_do_platform_check "${board//,/_}" "$1"
```

Comment:
Why is this shell pattern-matching needed here? It should not...

Reply:
@dangowrt I have yet to find where `$board_name` is set and since the same shell pattern-matching was used in `target/linux/bcm53xx/base-files/lib/upgrade/platform.sh` I did assume that this is an acceptable solution.

I do disagree with @ptpt52 in that his PR doesn't explain why it happens (to me at least it's not clear) but gave me a hint how to solve it.

A query in the forum showed an thread regarding `$board_name` but no replies.

Maybe you care to shortly explain or point me to where to find related documentation?

Reply:
I now had a closer look:

Assumption: `$(cat /tmp/sysinfo/board_name) == <vendor>,<model> == $board_name` (always true?)

Problem: nand_do_platform_check() in ./package/base-files/files/lib/upgrade/nand.sh fails to get control_length for Netgear WAX206 because is expects the sysupgrade folder to be named 'sysupgrade-$board_name'. E.g. 'sysupgrade-netgear,wax206' but the folder is named 'sysupgrade-netgear_wax206'.

nand_do_platform_check() is in use by six platforms:
- mediatek/mt7622:
  - rfb1-ubi: sysupgrade folder is indeed named 'sysupgrade-$board_name'
  - wax206: check fails
  - other: either no sysupgrade tar or folder is named `sysupgrade-<vendor>_<model>`
- mediatek/fitlogic : no sysupgrade file found in archive.openwrt.org
- bcm53xx: nand_do_platform_check() uses the same pattern matching for 'meraki,mr26' and 'meraki,mr32' and all other devices get handled by calling 'platform_other_check_image "$1"', without the $board_name parameter
- layerscape: use fix string 'traverse-ls1043' as board name
- pistachio: normal call to nand_do_platform_check and the sysupgrade folder in sysupgrade tar is named 'sysupgrade-img,pistachio-marduk'
- imx/cortexa9: the only sysupgrade file on archive.openwrt.org/.../targets/imx/cortexa9/ is for 'toradex,apalis*' and nand_do_platform_check() always 'returns 0' for these devices

Summary: Right now it's handled differently for different platforms. Implementing @ptpt52 PR will likely break it for `pistachio` and `mediatek/mt7622-*rfb1-ubi*` but will work for the rest.

Any advise @dangowrt ? Create a new issue for it on github?

Reply:
@ptpt52 I was wrong, you do check for both <vendor>_<model> and <vendor>,<model> it won't break.

It may be helpful to know if these two are the only variants that are to be expected. Because I can set BOARD_NAME in the *.mk file to whatever I like and that will be second part of the `sysupgrade-*` folder name.
As an example for the Xiaomi AX6S it set:

```
./target/linux/mediatek/image/mt7622.mk:  BOARD_NAME := xiaomi,redmi-router-ax6s
```

Reply:
sysupgrade on pistachio may be broken since a long time already.  The target is not used much.

Why don't we use `<vendor>,<model> ` in the sysupgrade images?

Reply:
In `include/image-commands.mk`

```
492 define Build/sysupgrade-tar
493     sh $(TOPDIR)/scripts/sysupgrade-tar.sh \
494         --board $(if $(BOARD_NAME),$(BOARD_NAME),$(DEVICE_NAME)) \
495         --kernel $(call param_get_default,kernel,$(1),$(IMAGE_KERNEL)) \
496         --rootfs $(call param_get_default,rootfs,$(1),$(IMAGE_ROOTFS)) \
497         $@
498 endef
```

if BOARD_NAME is not set (which seems for most devices) then DEVICE_NAME is used instead. It would seem that DEVICE_NAME is (mostly?) `<vendor>_<model>`.
The sysupgrade script in OpenWrt seems to uses /tmp/sysinfo/board_name which for the devices I got contains `<vendor>,<model>`. But then for them the image check function is only `return 0` and that "solves" that issue.

Reply:
@dangowrt, @hauke any advise how I should resolve that? Should I just set BOARD_NAME?

Reply:
Look at commits b3d699bdd3c and ec991424749b to understand how it is intended to work on modern targets with device tree for which the `compatible` string from device tree is used as `board_name`; setting `BOARD_NAME` is intended for legacy boards only.

@ptpt52 correctly pointed out that `platform_check_image()` simply return 0 for most targets which is most likely a work-around for that `',' != '_'` issue. Hence I believe that #11004 is the correct solution.

Reply:
Thank you @dangowrt for the explanation. We'll wait for #11004 to be merged to then maybe be able to merge this.